client/my-sites/checkout/composite-checkout/components/jetpack-pro-redirect-modal/index.tsx

```diff
@@ -49,6 +50,24 @@ export default function JetpackProRedirectModal( { redirectTo, productSourceFrom
 
 	const isAgencyPartner = useSelector( isAgencyUser );
 
+	// Function to record the event when the modal is displayed.
+	// It is in a separate useEffect to avoid unecessary re-renders.
+	useEffect( () => {
+		const recordShowEvent = () => {
+			dispatch( recordTracksEvent( 'jetpack_dashboard_agency_checkout_redirect_modal_show' ) );
+		};
+
+		if (
+			isAgencyPartner &&
+			! isDismissed &&
+			( productSourceFromUrl === 'jetpack-plans' ||
+				redirectURLPage === 'my-jetpack' ||
+				redirectURLPage === 'jetpack' )
+		) {
+			recordShowEvent();
+		}
+	}, [ isAgencyPartner, isDismissed, productSourceFromUrl, redirectURLPage, dispatch ] );
```

Comment:
Can we keep this empty? I think this will also run when `isDismissed` changes.

```suggestion
         // We only want to run this once
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, [] );
```

Reply:
ah ha!  Ok, the linter was throwing me because it was mad about the deps, But this way it will only run once like your comment says.